docker/Dockerfile

```diff
@@ -2,36 +2,26 @@
 # Install OpenSearch Benchmark to build a Docker image #
 ########################################################
 
-ARG VERSION
-ARG BUILD_ENV=production
-
-FROM python:3.11.2-slim as build_env_testing
-ONBUILD COPY opensearch-benchmark-git/ ./
-
-FROM python:3.11.2-slim as build_env_production
-ONBUILD RUN echo Production Environment
-
-FROM build_env_${BUILD_ENV}
+FROM python:3.11.2-slim
 WORKDIR /opensearch-benchmark
 ENV BENCHMARK_RUNNING_IN_DOCKER=True
 
 RUN apt-get -y update && \
-    apt-get install -y curl git gcc pbzip2 pigz && \
+    apt-get install -y curl git gcc pbzip2 pigz make jq && \
     apt-get -y upgrade && \
     rm -rf /var/lib/apt/lists/*
 
 RUN groupadd --gid 1000 opensearch-benchmark && \
     useradd -d /opensearch-benchmark -m -k /dev/null -g 1000 -N -u 1000 -l -s /bin/bash benchmark
 
-ENV PIP_ONLY_BINARY=h5py
-RUN if [ "$BUILD_ENV" = "testing" ] ; then echo Testing; ls -l; python3 -m pip install -e . ; \
-    else echo Production; if [ -z "$VERSION" ] ; then python3 -m pip install opensearch-benchmark ; else python3 -m pip install opensearch-benchmark==$VERSION ; fi; fi
+RUN set -e; git clone https://github.com/opensearch-project/opensearch-benchmark; cd opensearch-benchmark; make install; cd ..; rm -rf opensearch-benchmark
```

Comment:
If we really want to build from source, then we should introduce two stages of docker build.

Stage 1 do the build with tools like make/jq installed, while Stage 2 copy only the binaries / installed folders cleanly. With this current change it will increase the size of the docker layer, which increase the size of the docker image as a whole.

Reply:
I do, still prefer that the prod mode is using release pypi build, to be much simpler and clean, but retain test mode with the build from source approach, if make sense.

Reply:
Changed this to multi-stage, so the image size has gone down:
```
<none>                <none>        d294c275b627   2 hours ago      694MB

osb/osb-linux-amd64   latest        b49285646311   23 seconds ago   475MB
```

I think there's some value to keeping this independent of the PyPI build, so left that alone for now.  If it turns out to be an issue, it can be changed fairly easily.